api/rest/client/methods.go

```diff
@@ -172,3 +178,72 @@ func (c *Client) GetConnectGraph() (api.ConnectGraphSerial, error) {
 	err := c.do("GET", "/health/graph", nil, &graphS)
 	return graphS, err
 }
+
+// WaitFor is a utility function that allows for a caller to
+// wait for a paticular status for a CID. It returns a channel
+// upon which the caller can wait for the targetStatus.
+func (c *Client) WaitFor(ctx context.Context, ci *cid.Cid, local bool, target api.TrackerStatus, checkFreq time.Duration) (<-chan api.TrackerStatus, <-chan error) {
```

Comment:
I realize I wasn't super clear. The returned channel should provide `GlobalPinInfo` objects (this is the result of the `ctl status` calls and thus the confusion).